### PR TITLE
[MRG] Fix match function to perform matches based on suffix and datatype

### PR DESCRIPTION
**Pull request by adam2392**
PR Description
--------------

Closes: #568 

Also closes: #570 

Need to decide:

1. add suffix to BIDSPath.entities list
2. remove suffix from get_entities_from_fname, which would break a lot of things
3. Or adjust code inside match() as is done right now.

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

**Comment by agramfort** [comment]
tell us when not WIP anymore

**Comment by adam2392** [comment]
> tell us when not WIP anymore

It's good for the initial review now! There is still the matter of whether or not we include suffix in the `BIDSPath.entities`, or not, but everything else was fixed and updated w/ a unit test.

File: mne_bids/tests/test_path.py [hunk 1]
```diff
@@ -596,6 +596,45 @@ def test_match(return_bids_test_dir):
...
+    # Check handling of `extension` and `suffix`, part 1: suffix passed
+    bids_path_01 = BIDSPath(root=bids_root, run='01',
+                            suffix='channels', extension='.tsv',
+                            datatype='meg')
+    paths = bids_path_01.match()
+    assert len(paths) == 1
+    assert paths[0].extension == '.tsv'
+    assert paths[0].suffix == 'channels'
+
+    # Check handling of `datatype` when explicitly passe din
```
Comment (agramfort) [inline mne_bids/tests/test_path.py:615]: passe din?

**Review by agramfort — COMMENTED** [review]
@hoechenberger can you have a look?

nothing looks fishy at first sight

File: mne_bids/tests/test_path.py (hunk 1 above)
Comment (adam2392) [inline mne_bids/tests/test_path.py:615]:
```suggestion
    # Check handling of `datatype` when explicitly passed in
```

**Review by hoechenberger — APPROVED** [review]
I'm not a big fan of inferring things on so many levels as we do in MNE-BIDS, but aside from that, this LGTM

**Comment by hoechenberger** [comment]
Thanks @adam2392

**Pull request merged**